.github/actions/build-binaries/linux/Dockerfile

```diff
@@ -0,0 +1,105 @@
+# Modified from https://github.com/oleksis/pyinstaller-manylinux
+
+ARG BASE_IMAGE=quay.io/pypa/manylinux_2_28_x86_64
+FROM ${BASE_IMAGE} AS builder
+
+LABEL org.opencontainers.image.description="Run PyInstaller on ManyLinux 2.28 (AlmaLinux 8.7 based)"
+LABEL org.opencontainers.image.licenses=MIT
+
+SHELL ["/bin/bash", "-c"]
+
+ARG HOME=/root
+ARG PYTHON_VERSION=3.12
+ARG PYTHON_LAST=3.12.9
+ARG PYINSTALLER_VERSION=6.12.0
```

Comment:
This being hard coded here, and being version restricted but not hard coded in `pyproject.toml`, which is used for Windows and macOS, but not Linux, is a bit of a gotcha, no?

Reply:
I think I found a way around this in my branch, we can defer installing PyInstaller until we actually run the container, then we can install it using poetry

Reply:
@achidlow sounds good, once you'll finalize your changes on the pr it might be easier for you to retarget against this branch then merge it here (would also be easier to see the diffs on what your pr simplifies). And then we can merge this pr (or apply any remaining last minute feedback/tweaks)